new-packages/ts-project/__tests__/transitions.test.ts

```diff
@@ -0,0 +1,2235 @@
+                  "metaEntries": [],
+                  "tags": [],
+                  "type": "normal",
+                },
+                "parentId": "state-0",
+                "type": "node",
+                "uniqueId": "state-1",
+              },
+              "state-2": {
+                "data": {
+                  "description": undefined,
+                  "entry": [],
+                  "exit": [],
+                  "history": undefined,
+                  "initial": undefined,
+                  "invoke": [],
+                  "metaEntries": [],
+                  "tags": [],
+                  "type": "normal",
+                },
+                "parentId": "state-0",
+                "type": "node",
+                "uniqueId": "state-2",
+              },
+              "state-3": {
+                "data": {
+                  "description": undefined,
+                  "entry": [],
+                  "exit": [],
+                  "history": undefined,
+                  "initial": undefined,
+                  "invoke": [],
+                  "metaEntries": [],
+                  "tags": [],
+                  "type": "normal",
+                },
+                "parentId": "state-0",
+                "type": "node",
+                "uniqueId": "state-3",
+              },
+            },
+            "root": "state-0",
+          },
+          [],
+        ],
+      ]
+    `);
+});
+
+test('should extract transition with multiple targets (array of strings)', async () => {
+  const tmpPath = await testdir({
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { createMachine } from "xstate";
+
+      createMachine({
+        on: {
+          MULTIPLE: {
+            target: [".a.a2", ".b.b2"],
+          },
+        },
+        type: "parallel",
+        states: {
+          a: {
+            initial: "a1",
+            states: {
+              a1: {},
+              a2: {},
+            },
+          },
+          b: {
+            initial: "b1",
+            states: {
+              b1: {},
+              b2: {},
+            },
+          },
+        },
+      });
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+  expect(replaceUniqueIds(project.extractMachines('index.ts')))
+    .toMatchInlineSnapshot(`
+      [
+        [
+          {
+            "blocks": {},
+            "data": {
+              "context": {},
+            },
+            "edges": {
+              "edge-0": {
+                "data": {
+                  "actions": [],
+                  "description": undefined,
+                  "eventTypeData": {
+                    "eventType": "MULTIPLE",
+                    "type": "named",
+                  },
+                  "guard": undefined,
+                  "internal": true,
+                },
+                "source": "state-0",
+                "targets": [
+                  "state-3",
+                  "state-6",
```

Comment:
Why do we have both of these states as the target?

Reply:
The test title is:
> should extract transition with multiple targets (array of strings)

Those are two parallel regions that are targeted here by a single transition

Reply:
Is that how Studio behaves too?

Reply:
Quickly tested and this is not an edge Studio can handle. Honestly, multi target is a new concept to me.

Reply:
That's something to be fixed on the Studio's side.